### ISSUE TITLE
Consider supporting plugin-specific builders

### ISSUE THREAD

**Issue by alanking** [irods/irods_development_environment#69]
Instead of using a single plugin builder which calls the build hook script provided by each plugin, we should consider having each plugin provide its own builder Dockerfile which includes all of the required build dependencies. This would make the iteration cycle for developers much faster and avoid wasted downloads (which would occur only when rebuilding the image).

As a future enhancement, this may include having the development environment provide a "base image" of sorts for each supported platform to avoid some duplication.

**Comment by trel** [comment]
I really like the base image/dockerfile idea... but it would live in this repo, while the per-plugin ones would live... 'over there'.

Can per-plugin builder dockerfiles be built upon URL-defined base dockerfiles?

**Comment by alanking** [comment]
Not sure. I think you can only specify a base image using the `FROM` command, which would mean the base image would have to already have been built.

**Comment by alanking** [comment]
That's correct, @stefan-wolfsheimer. We would migrate (or duplicate?) the requirements which currently live in the build hook to a Dockerfile which could then be made into a builder image. This would of course require a script similar to the existing "core" builder, and that could be fetched from a URL as part of the image-building process. As long as the plugins are proper CMake projects it should be pretty straightforward to put together a generic way to build all of them (i.e. run `cmake`, run `make`, pass options as appropriate).
